### PR TITLE
Inject CLAUDE_CODE_OAUTH_TOKEN into per-patch env on macOS

### PR DESCRIPTION
**Pull request by subsetpark**
## Summary

- On macOS, Claude stores its OAuth credential in the Keychain rather than `.credentials.json`. Onton's per-patch `CLAUDE_CONFIG_DIR` scoping breaks the Keychain lookup, so every spawned `claude` reports `Not logged in · Please run /login` even when the user's shell session is logged in fine.
- Inject `CLAUDE_CODE_OAUTH_TOKEN` (precedence #5 in [Claude's auth docs](https://code.claude.com/docs/en/iam#authentication-precedence)) into the per-patch env when one is available — first from the parent env, then from `$XDG_CONFIG_HOME/onton/claude-oauth-token`. Users generate the token once via `claude setup-token`. Linux/Windows users are unaffected since file-based credentials still work for them.
- Added 7 unit tests for `resolve_claude_oauth_token_with`: env-var precedence, file fallback, whitespace stripping, XDG vs HOME, missing file, empty file, no-config case.
- Documented setup in README under a new "Coding-agent authentication" section.
- Left `seed_link` alone — its `src_exists` guard already prevents creating dangling symlinks; existing stale ones in `~/.local/share/onton/*/spawn-envs/*/claude/` are a one-time `find -L … -delete` cleanup, not a code-level concern.

## Test plan

- [x] `dune build` clean
- [x] `dune runtest` — all existing tests pass plus the 7 new ones
- [x] Format check passes (pre-commit hook)
- [ ] Reviewer to verify: with `~/.config/onton/claude-oauth-token` populated from `claude setup-token`, the rebuilt onton successfully spawns claude on macOS without the "Not logged in" error

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
<a href="https://app.blacksmith.sh/flowglad/codesmith/onton/pr/258"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-in-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-in-codesmith-light.svg">View in Codesmith</picture></a>
<sup>Need help on this PR? Tag <code>@codesmith</code> with what you need.</sup>

- [ ] Let Codesmith autofix CI failures and bot reviews

---
## Summary by cubic
Fixes macOS Claude auth in per-patch envs by resolving and injecting `CLAUDE_CODE_OAUTH_TOKEN` into spawned envs, preferring the parent env and falling back to `$XDG_CONFIG_HOME/onton/claude-oauth-token` (or `~/.config/...`). This stops spawned `claude` from showing “Not logged in” while keeping Linux/Windows behavior unchanged.

- **Bug Fixes**
  - Made token resolution self-contained in `spawn_env`: read from env first; else from config file; trims whitespace; XDG over HOME; safe on missing/empty files.
  - Injects the resolved token into per-patch `CLAUDE_CONFIG_DIR`; added unit tests and README guidance.

- **Migration**
  - Run `claude setup-token`, then save it to `~/.config/onton/claude-oauth-token` (mode 0600).
  - Or export `CLAUDE_CODE_OAUTH_TOKEN` in your shell.

<sup>Written for commit 682487298fc86ef3fffdcdc7ccb8622e5062fdbe. Summary will update on new commits.</sup>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -188,6 +188,37 @@ gh pr list --limit 1  # smoke-test repo access
...
+`CLAUDE_CODE_OAUTH_TOKEN` (precedence #5 in [Claude Code's credential
+docs](https://code.claude.com/docs/en/iam#authentication-precedence)).
+
+```sh
+# 1. Generate a 1-year token from your existing subscription (browser OAuth).
+claude setup-token
+
+# 2. Save it where onton will pick it up. Mode 0600 — it's a credential.
+mkdir -p ~/.config/onton
+umask 077 && printf %s "PASTE-TOKEN-HERE" > ~/.config/onton/claude-oauth-token
```
[the remark this answers is not in the document]
Comment (subsetpark) [reply README.md]: Fixed. The README now creates/truncates the token file with mode 0600, reads the token from stdin via cat instead of embedding it in the shell command, and reapplies chmod 600 after writing.

File: lib/spawn_env.ml [hunk 2]
```diff
@@ -41,6 +42,54 @@ let resolve_user_config_dir ~env_var ~home_subpath =
...
+let resolve_claude_oauth_token_with ~getenv_opt ~read_token_file =
+  match getenv_opt "CLAUDE_CODE_OAUTH_TOKEN" with
+  | Some t when not (String.is_empty (String.strip t)) -> None
+  | _ ->
+      let xdg_dir =
+        match getenv_opt "XDG_CONFIG_HOME" with
+        | Some d when not (String.is_empty d) -> Some d
+        | _ ->
+            Option.map (getenv_opt "HOME") ~f:(fun home ->
+                Stdlib.Filename.concat home ".config")
```
[the remark this answers is not in the document]
Comment (subsetpark) [reply lib/spawn_env.ml]: Fixed. The HOME fallback now treats an empty HOME the same as missing, so it no longer constructs a relative .config/onton/claude-oauth-token path. I also added an inline regression test for empty HOME.

File: lib/spawn_env.ml [hunk 3]
```diff
@@ -41,6 +42,54 @@ let resolve_user_config_dir ~env_var ~home_subpath =
...
       | _ -> None)
 
+(* On macOS, Claude Code stores its OAuth token in the macOS Keychain rather
+   than in [.credentials.json]. The Keychain lookup is scoped such that
+   pointing Claude at a per-patch [CLAUDE_CONFIG_DIR] makes it report
+   "Not logged in" even when the user has a valid Keychain entry. The
+   documented escape hatch is [CLAUDE_CODE_OAUTH_TOKEN] (precedence #5 in
+   docs), generated via [claude setup-token]. We read it from the shell env
+   (preferred) or fall back to [$XDG_CONFIG_HOME/onton/claude-oauth-token]
+   (or [~/.config/onton/claude-oauth-token]). *)
```
[the remark this answers is not in the document]
Comment (subsetpark) [reply lib/spawn_env.ml:52]: Fixed. resolve_claude_oauth_token_with now returns Some of the stripped CLAUDE_CODE_OAUTH_TOKEN when the parent env has a non-empty value, so claude_oauth_token_overrides is self-contained even with a sparse base_env. The inline env-var test now asserts that value is returned.

[the remark this answers is not in the document]
Comment (subsetpark) [reply lib/spawn_env.ml:52]: Agreed; this thread withdraws the issue and the existing full-parent-env call pattern was not dropping the token. I handled the separate self-contained override concern in the sibling thread.
